contracts/EscrowToken.sol

```diff
@@ -13,41 +13,66 @@ import { IRewarder } from "./interfaces/IRewarder.sol";
  * @author @cosmodude
  * @dev Implementation contract, instances are created as clones 
  */
-contract EscrowToken is IEscrow, IEscrowToken {
+contract EscrowToken is IEscrow {
   using SafeERC20 for IERC20;
 
   bool public initialized;
   address public quest;
   IERC20 public token;
+
+  uint32 public seekerId;
+  uint32 public solverId;
+
   uint256 public paymentAmount;
 
+  IRewarder public rewarder;
+
   modifier onlyQuest() {
     require(msg.sender == quest, "only quest");
     _;
   }
 
-  function initialize(address _token, uint256 _amount) external payable {   
+  function initialize(
+    address _token, 
+    address _rewarder,
+    uint32 _seekerId,
+    uint32 _solverId, 
+    uint256 _paymentAmount
+  ) external payable {   
     require(!initialized);
     require(_token != address(0), "Invalid token address");
+
     initialized = true;
+
     quest = msg.sender;
     token = IERC20(_token);
-    paymentAmount = _amount;
+    
+    seekerId = _seekerId;
```

Comment:
for ERC20 tokens, the tokens are transferred from the user straight to the escrow can be found in the ``startQuest()`` function

![image](https://github.com/theguild-quest/smart-contract/assets/52483029/e4889cd0-baed-4ed9-8d9e-8a1df32f7eae)

Reply:
This means that prior approval from the user is necessary

Reply:
oh, got it: the tokens go user -> escrow, never enter the quest

That's a good thing, since divides the roles right

Reply:
Yea, this is only possible for ERC20 tokens and not native tokens since we cant transfer native tokens on a user's behalf.

Not sure what you mean by dividing roles

Reply:
like quests should control the escrow and quest flow, not the funds